pkg/operator/ceph/object/rgw.go

```diff
@@ -331,16 +333,23 @@ func validateStore(context *clusterd.Context, s *cephv1.CephObjectStore) error {
 	// Validate the pool settings, but allow for empty pools specs in case they have already been created
 	// such as by the ceph mgr
 	if !emptyPool(s.Spec.MetadataPool) {
-		if err := pool.ValidatePoolSpec(context, s.Namespace, &s.Spec.MetadataPool); err != nil {
+		if err := pool.ValidatePoolSpec(r.context, s.Namespace, &s.Spec.MetadataPool); err != nil {
 			return errors.Wrap(err, "invalid metadata pool spec")
 		}
 	}
 	if !emptyPool(s.Spec.DataPool) {
-		if err := pool.ValidatePoolSpec(context, s.Namespace, &s.Spec.DataPool); err != nil {
+		if err := pool.ValidatePoolSpec(r.context, s.Namespace, &s.Spec.DataPool); err != nil {
 			return errors.Wrap(err, "invalid data pool spec")
 		}
 	}
 
+	// Fail if we detected an external CephCluster CR and the list of endpoints is empty
+	if r.cephClusterSpec.External.Enable {
+		if len(s.Spec.Gateway.ExternalRgwEndpoints) == 0 {
```

Comment:
@vbnrh Here is an example of a setting that could be checked in the admission controller